src/scripttabs.cpp

```diff
@@ -108,7 +108,7 @@ TYPE_SYMBOL asTypeTable[] =
 	{ "",	(INTERP_TYPE)ST_POINTER_STRUCTSTAT, AT_SIMPLE, NULL, NULL },	//for NULLSTRUCTURESTAT
 
 	/* This final entry marks the end of the type list */
-	{ "END OF TYPE LIST", (INTERP_TYPE)NULL,		AT_SIMPLE,	NULL,	NULL },
+	{ "END OF TYPE LIST", (INTERP_TYPE)0,		AT_SIMPLE,	NULL,	NULL },
```

Comment:
INTERP_TYPE is enum and has nothing to do with pointers, so there's no place for any kind of NULL here, with or without casts. This field is checked for 0 to detect end of the array, so this change is at least consistent.

The code was and still is broken though, as:
- one enum is cast to another one (SCR_USER_TYPES -> INTERP_TYPE) which may result in UB
- C arrays are used instead of C++ containers